app/views/spree/orders/_line_item.html.haml

```diff
@@ -22,7 +22,7 @@
   %td.text-right.cart-item-price{"data-hook" => "cart_item_price"}
     = line_item.single_display_amount_with_adjustments.to_html
   %td.text-center.cart-item-quantity{"data-hook" => "cart_item_quantity"}
-    = item_form.number_field :quantity, :min => 0, "ofn-on-hand" => variant.on_hand, "ng-model" => "line_item_#{line_item.id}", :class => "line_item_quantity", :size => 5
+    = item_form.number_field :quantity, :min => 0, "ofn-on-hand" => "#{variant.on_demand && 9999 || variant.on_hand}", "ng-model" => "line_item_#{line_item.id}", :class => "line_item_quantity", :size => 5
```

Comment:
We're displaying the on hand count as: `variant.on_demand && 9999 || variant.on_hand` ?

Reply:
no, this is not the display value, this is the on-hand value used to verify stock in JS. this is how it is done in JS when this ofn-on-hand directive is used...

Reply:
Can we use `Infinity` instead of 9999?

Reply:
hey @mkllnk we can, actually I went for it. this is the code change:
![Screenshot 2019-06-13 at 16 26 07](https://user-images.githubusercontent.com/1640378/59445715-0530c980-8df8-11e9-9e3c-38b4d42b4927.png)

BUT, this 9999 can actually be a good limit because if a crazy user tries to checkout 100k  items, the DB may crash with 100k inserts on spree_inventory_units and there will also be a number that it will make the code break somewhere for being too large...

I dont like the hard coded number 9999 at all but I think it's better than Infinity.
What do you think?

Reply:
Good thinking. The result is good but it's a mix of concerns. I'm okay with leaving it. Ideally we would then name the 9999 something like `item_order_limit`. It's probably logic that shouldn't be hidden in a view. But I don't want to stop the current PR. We can just leave it and act on it if it comes up again.

Reply:
yes, I agree. the good part of this is that it's so obviously wrong. yes, item_order_limit  should be in a helper or service.